tests/basic_integration/fns.sh

```
@@ -0,0 +1,138 @@
+}
+
+waiton() {
+    local file_path="$1"
+    until [ -f "$file_path" ]; do
+        sleep 1
+    done
+}
+
+waiton-files() {
+  local timeout=600  # 10 minutes timeout
+  local start_time=$(date +%s)
```

Shellcheck

[warning] Declare and assign separately to avoid masking return values.

(SC2155)